cmd/kubeadm/app/cmd/phases/certs.go

```diff
@@ -188,51 +189,51 @@ func getSANDescription(certSpec *certsphase.KubeadmCert) string {
 	return fmt.Sprintf("\n\nDefault SANs are %s", strings.Join(sans, ", "))
 }
 
-func addFlags(cmd *cobra.Command, cfgPath *string, cfg *kubeadmapiv1beta1.InitConfiguration, addAPIFlags bool) {
-	options.AddCertificateDirFlag(cmd.Flags(), &cfg.CertificatesDir)
-	options.AddConfigFlag(cmd.Flags(), cfgPath)
-	if addAPIFlags {
-		cmd.Flags().StringVar(&cfg.Networking.DNSDomain, "service-dns-domain", cfg.Networking.DNSDomain, "Alternative domain for services, to use for the API server serving cert")
-		cmd.Flags().StringVar(&cfg.Networking.ServiceSubnet, "service-cidr", cfg.Networking.ServiceSubnet, "Alternative range of IP address for service VIPs, from which derives the internal API server VIP that will be added to the API Server serving cert")
-		cmd.Flags().StringSliceVar(&cfg.APIServerCertSANs, "apiserver-cert-extra-sans", []string{}, "Optional extra altnames to use for the API server serving cert. Can be both IP addresses and DNS names")
-		cmd.Flags().StringVar(&cfg.APIEndpoint.AdvertiseAddress, "apiserver-advertise-address", cfg.APIEndpoint.AdvertiseAddress, "The IP address the API server is accessible on, to use for the API server serving cert")
+func runCAPhase(ca *certsphase.KubeadmCert) func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(certsData)
+		if !ok {
+			return errors.New("certs phase invoked with an invalid data struct")
+		}
+
+		if err := certsphase.CreateCACertAndKeyFiles(ca, data.Cfg()); err != nil {
+			return err
+		}
+
+		return nil
 	}
 }
 
-func makeCommandsForCA(ca *certsphase.KubeadmCert, certList []*certsphase.KubeadmCert, cfgPath *string, cfg *kubeadmapiv1beta1.InitConfiguration) []*cobra.Command {
-	subCmds := []*cobra.Command{}
+func makePhasesForCA(ca *certsphase.KubeadmCert, certList []*certsphase.KubeadmCert) []workflow.Phase {
```

Comment:
What about moving thos 5 lines into `getCertsSubPhases` L108, so we get almost all the UX stuff in one place?

Reply:
Done